adoc/extensions/sycl_khr_queue_empty_query.adoc

```diff
@@ -0,0 +1,80 @@
+[[sec:khr-queue-empty-query]]
+= sycl_khr_queue_empty_query
+
+This extension allows developers to query the queue's emptiness, meaning if all
+commands submitted to a queue have been completed.
+
+[[sec:khr-queue-empty-query-dependencies]]
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+[[sec:khr-queue-empty-query-feature-test]]
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_QUEUE_EMPTY_QUERY# to one of the values defined in the table
+below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+
+[[sec:khr-queue-empty-query-funct]]
+== New Queue Function to Query Emptiness
+
+This extension adds the following function to the [code]#sycl::queue# class,
+which provides information about the emptiness of the queue.
+
+'''
+
+.[apidef]#queue::khr_empty#
+[source,role=synopsis,id=api:queue-khr-empty]
+----
+bool khr_empty() const
+----
+
+_Returns:_ [code]#true# if all <<command,commands>> enqueued on this queue have
+completed, [code]#false# otherwise.
+
+{note} Since the implementation executes commands asynchronously, the returned
+value is a snapshot in time.
+{endnote}
+
+'''
+
+[[sec:khr-queue-empty-query-example]]
+== Example
+
+The example below demonstrates the usage of this extension.
+
+[source,,linenums]
+----
+#include <iostream>
+#include <algorithm>
+#include <sycl/sycl.hpp>
+
+int main() {
+  // Pool of queues
+  std::vector<sycl::queue> Qs;
+  size_t n_queues = 12;
+  for (size_t i = 0; i < n_queues; i++)
+    Qs.push_back(sycl::queue(sycl::property_list{sycl::property::queue::in_order{}}));
+
+  // Useful receipt for load-balancing
```

Comment:
```suggestion
  // Useful recipe for load-balancing
```

The word "receipt" doesn't make sense to me.  Did you mean "recipe"?

Reply:
Indeed :(